tests/contrib/celery/test_old_style_task.py

```diff
@@ -25,11 +26,27 @@ def run(self, *args, **kwargs):
                     return
                 CelerySubClass.apply_async(args=[], kwargs={"stop": True})
 
-        @patch_task_with_pin(pin=self.pin)
         class CelerySubClass(CelerySuperClass):
             pass
 
         t = CelerySubClass()
-        t.run()
-        spans = self.tracer.writer.pop()
-        self.assertEqual(len(spans), 2)
+        res = t.apply()
```

Comment:
Test changed to use `apply()` instead of `run()` because of the considerations above.